This add -fno-stack-protector to all compilation, since -f-stack-protector is currently unsupported by ROCm
Reference: https://bugs.gentoo.org/890377
index 465187e..0baf925 100644
--- a/src/comgr-compiler.cpp
+++ b/src/comgr-compiler.cpp
@@ -850,6 +850,8 @@ amd_comgr_status_t AMDGPUCompiler::processFile(const char *InputFilePath,
     Argv.push_back(Arg);
   }
 
+  Argv.push_back("-fno-stack-protector");
+
   for (auto &Option : ActionInfo->getOptions()) {
     Argv.push_back(Option.c_str());
     if (Option.rfind("--rocm-path", 0) == 0) {
